include/kafka/addons/KafkaMetrics.h

```diff
@@ -0,0 +1,208 @@
+#pragma once
+
+#include "kafka/Project.h"
+
+// https://github.com/Tencent/rapidjson/releases/tag/v1.1.0
+#include "rapidjson/document.h"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/writer.h"
+
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+
+
+namespace KAFKA_API {
+
+/**
+ * \brief Helps to parse the metrics string with JSON format.
+ */
+class KafkaMetrics
+{
+public:
+    /**
+     * \brief Initilize with the metrics string.
+     */
+    explicit KafkaMetrics(std::string jsonMetrics);
+
+    static const constexpr char* WILDCARD = "*";
+
+    using KeysType = std::vector<std::string>;
+
+    /**
+     * \brief The matched keys (for wildcards) and the value.
+     */
+    template<typename ValueType>
+    using ResultsType = std::vector<std::pair<KeysType, ValueType>>;
+
+    /**
+     * \brief Get integer value(s) for the specified metrics.
+     * Note: the wildcard ("*") is supported.
+     */
+    ResultsType<std::int64_t> getInt(const KeysType& keys) { return get<std::int64_t>(keys); }
+
+    /**
+     * \brief Get string value(s) for the specified metrics.
+     * Note: the wildcard ("*") is supported.
+     */
+    ResultsType<std::string> getString(const KeysType& keys) { return get<std::string>(keys); }
+
+    static std::string toString(const KafkaMetrics::KeysType& keys);
+
+    template<typename ValueType>
+    static std::string toString(const KafkaMetrics::ResultsType<ValueType>& results);
+
+private:
+    template<typename ValueType>
+    ResultsType<ValueType> get(const KeysType& keys);
+
+    template<typename ValueType>
+    void getResults(ResultsType<ValueType>&               results,
+                    KeysType&                             keysForWildcards,
+                    rapidjson::Value::ConstMemberIterator iter,
+                    KeysType::const_iterator              keysToParse,
+                    KeysType::const_iterator              keysEnd);
+
+    template<typename ValueType>
+    static ValueType getValue(rapidjson::Value::ConstMemberIterator iter);
+
+#if COMPILER_SUPPORTS_CPP_17
+    std::string         _decodeBuf;
```

Comment:
It's a pity that we can't use the `std::string` for C++11.

https://en.cppreference.com/w/cpp/string/basic_string/data
`std::basic_string<CharT,Traits,Allocator>::data`
* `const CharT* data() const noexcept; `  -- since C++11
* `CharT* data() noexcept;`  -- since C++17